drivers/platform/maxim/maxim_gpio.c

```diff
@@ -0,0 +1,439 @@
+	uint32_t stat_reg = gpio_regs->int_stat;
+
+	/** Clear interrupt flags */
+	gpio_regs->int_clr = stat_reg;
+	while(stat_reg) {
+		if(stat_reg & 1) {
+			if(!gpio_callback[pin])
+				return;
+			void *ctx = gpio_callback[pin]->ctx;
+			gpio_callback[pin]->callback(ctx, pin, NULL);
+		}
+		pin++;
+		stat_reg >>= 1;
+	}
+}
+
+/**
+ * @brief Obtain the GPIO decriptor.
+ * @param desc - The GPIO descriptor.
+ * @param param - GPIO initialization parameters
+ * @return 0 in case of success, errno error codes otherwise.
+ */
+int32_t max_gpio_get(struct gpio_desc **desc,
+		     const struct gpio_init_param *param)
+{
+	if(!param || param->number >= N_PINS)
+		return -EINVAL;
+
+	struct gpio_desc *descriptor = calloc(1, sizeof(*descriptor));
+	if(!descriptor)
+		return -ENOMEM;
+
+	descriptor->number = param->number;
+	descriptor->platform_ops = param->platform_ops;
+	descriptor->extra = param->extra;
+	((gpio_cfg_t *)descriptor->extra)->mask = BIT(param->number);
+
+	GPIO_Config(descriptor->extra);
+	*desc = descriptor;
+
+	return 0;
+}
+
+/**
+ * @brief Get the value of an optional GPIO.
+ * @param desc - The GPIO descriptor.
+ * @param param - GPIO initialization parameters
+ * @return 0 in case of success, errno error codes otherwise.
+ */
+int32_t max_gpio_get_optional(struct gpio_desc **desc,
+			      const struct gpio_init_param *param)
+{
+	if(param == NULL) {
+		*desc = NULL;
+		return 0;
+	}
+
+	return max_gpio_get(desc, param);
+}
+
+/**
+ * @brief Free the resources allocated by gpio_get().
+ * @param desc - The GPIO descriptor.
+ * @return 0 in case of success, errno error codes otherwise.
+ */
+int32_t max_gpio_remove(struct gpio_desc *desc)
+{
+	if(!desc)
+		return -EINVAL;
+
+	max_gpio_unregister_callback(desc->number);
+	free(desc);
+
+	return 0;
+}
+
+/**
+ * @brief Enable the input direction of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @return 0 in case of success, errno error codes otherwise.
+ */
+int32_t max_gpio_direction_input(struct gpio_desc *desc)
+{
+	if(!desc || !desc->extra || desc->number >= N_PINS)
+		return -EINVAL;
+
+	gpio_cfg_t *maxim_extra = (gpio_cfg_t *)desc->extra;
+
+	if(maxim_extra->port >= N_PORTS)
+		return -EINVAL;
+
+	maxim_extra->mask = BIT(desc->number);
+	maxim_extra->func = GPIO_FUNC_IN;
+	GPIO_Config(maxim_extra);
+
+	return 0;
+}
+
+/**
+ * @brief Enable the output direction of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @param value - The value.
+ *                Example: GPIO_HIGH
+ *                         GPIO_LOW
+ * @return 0 in case of success, errno error codes otherwise.
+ */
+int32_t max_gpio_direction_output(struct gpio_desc *desc, uint8_t value)
+{
+	if(!desc || !desc->extra || desc->number >= N_PINS)
+		return -EINVAL;
+
+	gpio_cfg_t *maxim_extra = (gpio_cfg_t *)desc->extra;
+
+	maxim_extra->mask = BIT(desc->number);
+	maxim_extra->func = GPIO_FUNC_OUT;
+	GPIO_Config(maxim_extra);
+
+	if(value == 0)
```

Comment:
nitpick: we can use `GPIO_LOW` in the if compare. Also we need a space between if and brackets e.g. `if (condition)`

Reply:
If we have `GPIO_HIGH_Z` option in `max_gpio_set_value` should we have it here as well?